### ISSUE TITLE
Rect2 * Transform2D broken in 3.x LTS for C#

### ISSUE THREAD

**Issue by andy-noisyduck** [godotengine/godot#86415]
### Tested versions

Reproducible in 3.x. Fixed in 4.x.

### System information

Godot 3.5 Mono

### Issue description

Multiplying a Rect2 by a transform gives incorrect results (see example steps). The inverse is also incorrect.

Looks like this was introduced by https://github.com/godotengine/godot/pull/52666 with the 3.x backport https://github.com/godotengine/godot/pull/52762. Master waster later fixed by https://github.com/godotengine/godot/pull/64729, but this fix was never backported.

I note the comment by [aaronfranke](https://github.com/aaronfranke) about sneaking the fix into the PR for 4.x rather than a separate PR (https://github.com/godotengine/godot/pull/64729#discussion_r951709835) 👀

### Steps to reproduce

```cs
var r = new Rect2(0, 0, 100, 100);
var foo = r * new Transform2D( 1, 0, 0, 1, 0, 0);  // = ((0, 0), (100, 100)) - OK
var bar = r * new Transform2D(-1, 0, 0, 1, 0, 0);  // = ((-100, 0), (200, 100)) - Incorrect
```

### Minimal reproduction project (MRP)

N/A

**Comment by raulsntos** [comment]
Thanks for the detailed report. Since you've already tracked down the cause of the issue, are you interested in submitting a PR with the fix? It's fine if you can't or don't want to, someone else can do it, I just want to make sure that I don't step on your toes.

**Comment by andy-noisyduck** [comment]
> Thanks for the detailed report. Since you've already tracked down the cause of the issue, are you interested in submitting a PR with the fix? It's fine if you can't or don't want to, someone else can do it, I just want to make sure that I don't step on your toes.

I can make a PR. It's just a one-line fix for both the operators. I've not done a backport - should I just target the current 3.5 branch with the PR?

**Comment by raulsntos** [comment]
For Godot 3 PRs target the [`3.x`](https://github.com/godotengine/godot/tree/3.x) branch.

**Comment by akien-mga** [comment]
Fixed by #86466.
